lib/engine/game/g_18_zoo/step/assign.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Engine
+  module Game
+    module G18ZOO
+      module Step
+        class Assign < Engine::Step::Assign
+          def available_hex(entity, hex)
+            return unless entity.company?
+            return unless entity.owner&.corporation?
+            return unless entity.owner.tokens.map { |token| token&.city&.hex&.id }.include?(hex.id)
```

Comment:
tokens.any? { |token| token&.city&.hex == hex )

Reply:
done in d493f74b4